iocage/helpers_ioctl.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2017-2018, Stefan Grönke
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted providing that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+# IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Interface for ioctl abstraction."""
+import socket
+import struct
+import fcntl
+import ipaddress
+
+from enum import Enum
+
+
+class SOCKIO_IOCTLS(Enum):
+    """Hardcoded ioctl numbers."""
+
+    SIOCGIFADDR = -1071617759
+    SIOCGIFMTU = -1071617741
```

Comment:
is there any reason you're not doing this thru `ctypes`, like this library does:

https://github.com/pydron/ifaddr/blob/master/ifaddr/_shared.py

?

Reply:
Yep, tested in on all available FreeBSD and HardenedBSD variants and the values are unchanged and functional. If they break, then in new releases - but errors will be noticed quickly when starting the next jail with VNET.